File: origin/perpendicular_geometic_factor.m
function [I_per] = perpendicular_geometic_factor(x1,x2,yb,xb,y1,y2,d)
%   x1              x2
%  yb o---------------o     
%         /d                 
%     xb /                 
%       o y1                
%       |             
%       |
%       |
%       |
%       |
%       |
%       |
%       o y2

f = inline('1./sqrt(x.^2+y.^2+z.^2)','x','y','z');%tr??c s?a l?i
I_per = dblquad(f,x1-xb,x2-xb,y1-yb,y2-yb,[],[],d);%tr??c s?a l?i

File: perpendicular_geometic_factor.py
import numpy as np
from scipy.integrate import dblquad

def perpendicular_geometic_factor(x1, x2, yb, xb, y1, y2, d):

    # integrand: 1/sqrt(x^2 + y^2 + d^2)
    def f(y, x):  # dblquad expects f(y,x)
        return 1.0 / np.sqrt(x**2 + y**2 + d**2)

    # integration bounds: shift coordinates by xb, yb
    I_per, _ = dblquad(f, x1 - xb, x2 - xb, lambda x: y1 - yb, lambda x: y2 - yb)
    return I_per


